File: src/main/java/com/myimages/service/PromptConfigService.java
package com.myimages.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.myimages.domain.PromptConfig;
import com.myimages.dto.PromptConfigDTO;
import com.myimages.dto.PromptConfigSearchDTO;
import com.myimages.dto.PromptConfigPageDTO;
import com.myimages.dto.PromptConfigConvertCriteriaDTO;
import com.myimages.service.GenericService;
import com.myimages.dto.common.RequestDTO;
import com.myimages.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface PromptConfigService extends GenericService<PromptConfig, Integer> {

	List<PromptConfig> findAll();

	ResultDTO addPromptConfig(PromptConfigDTO promptConfigDTO, RequestDTO requestDTO);

	ResultDTO updatePromptConfig(PromptConfigDTO promptConfigDTO, RequestDTO requestDTO);

    Page<PromptConfig> getAllPromptConfigs(Pageable pageable);

    Page<PromptConfig> getAllPromptConfigs(Specification<PromptConfig> spec, Pageable pageable);

	ResponseEntity<PromptConfigPageDTO> getPromptConfigs(PromptConfigSearchDTO promptConfigSearchDTO);
	
	List<PromptConfigDTO> convertPromptConfigsToPromptConfigDTOs(List<PromptConfig> promptConfigs, PromptConfigConvertCriteriaDTO convertCriteria);

	PromptConfigDTO getPromptConfigDTOById(Integer promptConfigId);


}

File: myimages_ui/src/services/PromptConfigService.js
import http from "../http-common"; 

class PromptConfigService {
  getAllPromptConfigs(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/promptConfig/promptConfigs`, searchDTO);
  }
 

  get(promptConfigId) {
    return this.getRequest(`/promptConfig/${promptConfigId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/promptConfig?field=${matchData}`, null);
  }

  addPromptConfig(data) {
    return http.post("/promptConfig/addPromptConfig", data);
  }

  update(data) {
  	return http.post("/promptConfig/updatePromptConfig", data);
  }
  
  uploadImage(data,promptConfigId) {
  	return http.postForm("/promptConfig/uploadImage/"+promptConfigId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new PromptConfigService();
